core2/pkg/accounting/accounting_internal.go

```diff
@@ -390,6 +391,124 @@ func internalCommitAllocation(b *internalBucket, allocId accAllocId) {
 	b.Mu.Unlock()
 }
 
+func internalUpdateAllocation(actor rpc.Actor, now time.Time, b *internalBucket, allocationId accAllocId, newQuota util.Option[int64], newStart util.Option[fndapi.Timestamp], newEnd util.Option[fndapi.Timestamp]) *util.HttpError {
+	b.Mu.Lock()
+	defer b.Mu.Unlock()
+
+	reference := actor.Username
+	if actor.Project.Present {
+		reference = string(actor.Project.Value)
```

Comment:
Need to check if role in project is admin or PI. Use the following snippet for this:

```go
actor.Membership[actor.Project.Value].Satisfies(rpc.ProjectRoleAdmin)
```